.travis.yml

```diff
@@ -1,77 +1,93 @@
 sudo: required
-
 services:
     - docker
 
-language: c
+env:
+    global:
+        - AUTHOR_EMAIL="$(git log -1 $TRAVIS_COMMIT --pretty=\"%aE\")"
+        - CI_TOOLS="$TRAVIS_BUILD_DIR/travis-ci/tools"
+
+stages:
+    # Run Coverity periodically instead of for each commit/PR
+    - name: Coverity
+      if: type = cron
 
 jobs:
     include:
-        - stage: coverity scan
-          before_script:
-              - sudo apt-get update
+        - stage: Build & test
+          name: Fedora Rawhide
+          language: bash
+          env:
+              - FEDORA_RELEASE="rawhide"
+              - CONT_NAME="systemd-fedora-$FEDORA_RELEASE"
+              - BUILD_REQS_FILE="$TRAVIS_BUILD_DIR/travis-ci/fedora-build.reqs"
+              - DOCKER_EXEC="docker exec -ti $CONT_NAME"
+          before_install:
               - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
               - docker --version
-              - env > .env
+          install:
+              # Pull a Docker image and start a new container
+              - docker pull fedora:$FEDORA_RELEASE
+              - docker run -v $TRAVIS_BUILD_DIR:/build:rw
+                           -v /var/lib/dbus/machine-id:/etc/machine-id:ro
+                           -w /build --privileged=true --name $CONT_NAME
+                           -dit --net=host fedora:$FEDORA_RELEASE /sbin/init
+              # Upgrade system and install necessary build/test requirements
+              - $DOCKER_EXEC dnf -y --refresh upgrade
+              - $DOCKER_EXEC dnf -y install $(cat $BUILD_REQS_FILE)
+          script:
+              - set -e
+              # Build systemd
+              - $DOCKER_EXEC meson build
+              - $DOCKER_EXEC ninja -C build
+              # Run 'make check'
+              - $DOCKER_EXEC ninja -C build test
+              - set +e
+
+        - stage: Coverity
```

Comment:
I think it would make sense to use something more stable than rawhide for Coverity.

Reply:
Yeah, that was something I wanted to discuss. I guess using `fedora:latest`, which points to a latest, let's say *stable*, release, should be fine.